### ISSUE TITLE
Symbols nupkgs should not be published in source-build

### ISSUE THREAD

**Issue by NikolaMilosavljevic** [dotnet/source-build#4453]
There are various *symbols*.nupkg files in source-build artifacts dir that should not be present. One example is: `Microsoft.CodeAnalysis.Workspaces.MSBuild.BuildHost.Symbols.x.y.z.nupkg`.

Most of symbols nupkgs are excluded in arcade publishing infra: https://github.com/dotnet/arcade/blob/7507f80c8db285bbc9939c1dff522a761cf4edc0/src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj#L89-L90 and https://github.com/dotnet/arcade/blob/7507f80c8db285bbc9939c1dff522a761cf4edc0/src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj#L138-L139

However, some uncommonly-named packages, like the one listed above are still "published" by repos and end up in source-build artifacts.

**Comment by mmitche** [comment]
I don't think you need to exclude those in Publish.proj. I think it would be enough to remove them from publishing in roslyn's [Publishing.props](https://github.com/dotnet/roslyn/blob/main/eng/Publishing.props) under source-only mode.

**Comment by MichaelSimons** [comment]
@NikolaMilosavljevic, Can a test be written to catch these instances going forward?

**Comment by NikolaMilosavljevic** [comment]
> I don't think you need to exclude those in Publish.proj. I think it would be enough to remove them from publishing in roslyn's [Publishing.props](https://github.com/dotnet/roslyn/blob/main/eng/Publishing.props) under source-only mode.

Yes, absolutely. I will just need to add another target similar to this one, for source-only case - https://github.com/dotnet/roslyn/blob/d26e7278f0d7d75f9ad8810d957de2bbb802e8ea/eng/Publishing.props#L21-L24

**Comment by NikolaMilosavljevic** [comment]
This is actually rather odd. The only reason we currently exclude `*.symbols.nupkg` is due to `Distinct()` in the task that merges the artifacts from repo manifests: https://github.com/dotnet/dotnet/blob/e14603a2e01174a9e295475b05c4afc4457112bc/eng/tools/tasks/Microsoft.DotNet.UnifiedBuild.Tasks/GetKnownArtifactsFromAssetManifests.cs#L67

It seems that repos are still publishing symbols packages, even in source-only build, and they end up in `artifacts/packages` and in the manifest, so for instance we'll have these two packages from `runtime`:

```
<Package Id="Microsoft.Bcl.AsyncInterfaces" Version="10.0.0-dev" DotNetReleaseShipping="true" PipelineArtifactName="linux_x64_Artifacts" PipelineArtifactPath="artifacts/packages/Release/Shipping/runtime/Microsoft.Bcl.AsyncInterfaces.10.0.0-dev.nupkg" RepoOrigin="runtime" Visibility="External" />
<Package Id="Microsoft.Bcl.AsyncInterfaces" Version="10.0.0-dev" DotNetReleaseShipping="true" PipelineArtifactName="linux_x64_Artifacts" PipelineArtifactPath="artifacts/packages/Release/Shipping/runtime/Microsoft.Bcl.AsyncInterfaces.10.0.0-dev.symbols.nupkg" RepoOrigin="runtime" Visibility="External" />
```

Due to the code above, the second package gets discarded late in the build, in https://github.com/dotnet/dotnet/blob/e14603a2e01174a9e295475b05c4afc4457112bc/eng/Publishing.props#L48

I'm not happy that filtering of `*.symbols.nupkg` seems "accidental" or at least a side-effect.

To filter `roslyn`'s oddly-named symbols packages, I can add filtering to VMR's `Publishing.props` or VMR's `PublishSourceBuild.props` i.e.: https://github.com/dotnet/dotnet/blob/e14603a2e01174a9e295475b05c4afc4457112bc/eng/PublishSourceBuild.props#L91

Questions:
- Should we update `arcade`'s Publish.proj, for source-only build, to filter all symbols packages, or at least `*.symbols.nupkg` so they don't get copied to VMR packages location and don't get included in repo manifest?
- Should we filter `roslyn`'s `*Symbols*` packages in VMR's publishing targets or in `roslyn` repo's Publishing.props?

**Comment by mmitche** [comment]
Do you see the `.symbols.nupkg` behavior from all repos, or just from a few (e.g. runtime and roslyn). We might be seeing the custom publishing logic from a few repos causing some set of symbol packages to show up.

In any case, I think it does make sense that we exclude standard patterns from Publish.proj. That would just be .symbols.nupkg. I would put roslyn's specially named symbol packages in their Publishing.props

**Comment by NikolaMilosavljevic** [comment]
`.symbols.nupkg` from all repos are published in source-only build.

I'll update `arcade`'s `Publish.proj` to fix this for standard symbols packages and add custom exclusion in `roslyn`'s `Publishing.props`.
